tests/CustomTest.php

```diff
@@ -19,14 +22,19 @@ function init()
 
         $this->addHook('beforeSave', function($m) {
             if ($m->isDirty('gender')) {
-                $m->audit_log['action'] = 'genderbending';
+                //$m->audit_log['action'] = 'genderbending'; // deprecated usage
+                //$m->audit_log_controller->audit_log_stack[0]['action'] = 'genderbending';
+                $m->audit_log_controller->custom_action = 'genderbending';
```

Comment:
can this be done through a method call? 

$m->addHistoryComment("genderbending");

Reply:
well - if that's in beforeSave hook, then it already have done audit->push() and set action='update'. so to change that at this point you have only one option - set `controller->custom_action` and/or `controller->custom_fields` properties which are respected in audit->pull().

I intentionally didn't want developer to mess directly with audit data model. He should use audit controller for that and audit controller should take care of audit data model.